planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp

```diff
@@ -1586,14 +1582,18 @@ std::vector<geometry_msgs::msg::Point> postProcess(
       calcLongitudinalOffsetGoalPoint(tmp_bound, goal_pose, goal_start_idx, vehicle_length);
     const auto p_tmp =
       geometry_msgs::build<Pose>().position(goal_point).orientation(goal_pose.orientation);
-    const size_t goal_idx = std::max(
-      goal_start_idx, findNearestSegmentIndexFromLateralDistance(tmp_bound, p_tmp, M_PI_2));
-
+    const size_t goal_nearest_idx =
+      findNearestSegmentIndexFromLateralDistance(tmp_bound, p_tmp, M_PI_2);
+    const size_t goal_idx = ((goal_start_idx - start_idx) * (goal_start_idx - start_idx) >
+                             (goal_nearest_idx - start_idx) * (goal_nearest_idx - start_idx))
+                              ? goal_start_idx
+                              : goal_nearest_idx;
     return std::make_pair(goal_idx, goal_point);
   }();
 
   // Insert middle points
-  for (size_t i = start_idx + 1; i <= goal_idx; ++i) {
+  size_t step = (start_idx < goal_idx) ? 1 : -1;
+  for (size_t i = start_idx + step; i != goal_idx + step; i += step) {
```

Comment:
Previously,  inseting the points from `start_idx` to `goal_idx` didn't consider `start_idx > goal_idx`  (is_driving_forward=false). This considers that condition.
 


2. Fix visualizeBound in tier4_planning_rviz_plugin/display.hpp